test/utils/shallowEquality.js

```diff
@@ -0,0 +1,68 @@
+import expect from 'expect';
+import shallowEqualScalar from '../../src/utils/shallowEqualScalar';
+import shallowEqual from '../../src/utils/shallowEqual';
+
+describe('Utils', () => {
+  describe('shallowEqualScalar', () => {
+    it('should compare the given arguments shallowly', () => {
+
+      expect(shallowEqualScalar(null, undefined)).toBe(false);
+      expect(shallowEqualScalar(undefined, null)).toBe(false);
+
+      expect(shallowEqualScalar([], [])).toBe(true);
+      expect(shallowEqualScalar([], [1, 2])).toBe(false);
+      expect(shallowEqualScalar([3, 4], [3, 4])).toBe(true);
+      expect(shallowEqualScalar([3, 4], [3, 5])).toBe(false);
+      expect(shallowEqualScalar([[]], [[]])).toBe(false);
+
+      const m = {one: 1};
+      const n = {two: 2};
+
+      expect(shallowEqualScalar(m, n)).toBe(false);
+
+      const a = {
+        one: 1,
+        two: 2
+      };
+
+      const b = a;
+      b.three = 3;
+
+      expect(shallowEqualScalar(a, a)).toBe(true);
+      expect(shallowEqualScalar(a, b)).toBe(true);
+
+    });
+  });
+
+  describe('shallowEqual', () => {
+    it('should compare the given objects shallowly', () => {
+
+      // expect(shallowEqual(null, undefined)).toBe(true);
```

Comment:
Here, too, we should assume `typeof` of whatever passed is `object`. This includes `null`, but not `undefined`.